addons/binding/org.openhab.binding.denonmarantz/src/main/java/org/openhab/binding/denonmarantz/internal/DenonMarantzConnector.java

```diff
@@ -0,0 +1,785 @@
+        this.httpClient = new HttpClient();
+    }
+
+    public DenonMarantzState getState() {
+        return state;
+    }
+
+    /**
+     * Set up a telnet connection to the receiver AND fetch initial state over HTTP.
+     */
+    public void connect() {
+        if (config.isTelnet()) {
+            telnetClient = new DenonMarantzTelnetClient(config, this);
+            telnetClient.start();
+        } else {
+            // If Telnet is disabled, schedule polling the AVR using HTTP
+            refreshState();
+            startPolling();
+        }
+    }
+
+    private void startPolling() {
+        if (!isPolling()) {
+            logger.info("HTTP polling started.");
+            pollingJob = scheduler.scheduleAtFixedRate(new Runnable() {
+                @Override
+                public void run() {
+                    refreshHttpProperties();
+
+                }
+            }, config.httpPollingInterval, config.httpPollingInterval, TimeUnit.MILLISECONDS);
+        }
+    }
+
+    private boolean isPolling() {
+        return pollingJob != null && !pollingJob.isCancelled();
+    }
+
+    private void stopPolling() {
+        if (pollingJob != null) {
+            pollingJob.cancel(true);
+            logger.info("HTTP polling stopped.");
+        }
+    }
+
+    @Override
+    public void telnetClientConnected(boolean connected) {
+        if (!connected) {
+            if (config.isTelnet() && !disposing) {
+                logger.debug("Telnet client disconnected.");
+                stateListener.connectionError(
+                        "Error connecting to the telnet port. Consider disabling telnet in this Thing's configuration to use HTTP polling instead.");
+            }
+        }
+        refreshState();
+    }
+
+    /**
+     * Shutdown the telnet client (if initialized) and the http client
+     */
+    public void dispose() {
+        logger.debug("disposing connector");
+        disposing = true;
+
+        stopPolling();
+
+        if (telnetClient != null) {
+            telnetClient.shutdown();
+        }
+
+        if (!httpClient.isStopped()) {
+            try {
+                httpClient.stop();
+            } catch (Exception e) {
+                logger.debug("Error stopping http client", e);
+            }
+        }
+    }
+
+    public void sendMuteCommand(Command command, int zone) throws UnsupportedCommandTypeException {
+        if (zone < 1 || zone > 3) {
+            throw new UnsupportedCommandTypeException("Zone must be in range [1-3], zone: " + zone);
+        }
+        String zonePrefix = zone == 1 ? "" : "Z" + zone;
+        String cmd = zonePrefix + "MU";
+        if (command == OnOffType.ON) {
+            cmd += "ON";
+        } else if (command == OnOffType.OFF) {
+            cmd += "OFF";
+        } else if (command instanceof RefreshType) {
+            if (!config.isTelnet()) {
+                // refreshing does not work using HTTP command, it will be polled soon.
```

Comment:
By default polling is done every 5 seconds, also with HTTP the refresh is done for ALL items as it fetches and processes an xml with the receiver state.
I've observed that upon start of OpenHAB all items are being refreshed. That would result in a lot of requests being executed "in between" the regular interval (probably these will all run in parallel and might even cause issues at the end of the receiver).

You are right about the need to factor this out, will do that.

Reply:
> I've observed that upon start of OpenHAB all items are being refreshed

That is true. Upon the OH2 start a `Refresh` command is send to all items. For http this should be fine. But if you are saying that this would be a problem for the telnet connection because you will have a new poll attempt for each item in a very short time frame, then I guess we have to live with the fact that we have to wait until the normal polling fetches the data.